vowpalwabbit/learner.h

```diff
@@ -205,7 +274,9 @@ struct learner
     assert((is_multiline && std::is_same<multi_ex, E>::value) ||
         (!is_multiline && std::is_same<example, E>::value));  // sanity check under debug compile
     increment_offset(ec, increment, i);
+    log_entry(ec, "learn");
```

Comment:
this call and log_exit would still happen if you disable VW_DBG

Reply:
All the statements inside the function get compiled out if constants are false.